library/std/src/env.rs

```diff
@@ -188,12 +188,8 @@ impl fmt::Debug for VarsOs {
 /// Errors if the environment variable is not present.
 /// Errors if the environment variable is not valid Unicode. If this is not desired, consider using
 /// [`var_os`].
-///
-/// # Panics
-///
-/// This function may panic if `key` is empty, contains an ASCII equals sign
-/// `'='` or the NUL character `'\0'`, or when the value contains the NUL
-/// character.
+/// May error if the `key` is empty, contains an ASCII equals sign `'='`, or contains the NUL character `'\0'`.
+/// May error when the value contains the NUL character.
```

Comment:
Are there any platforms where this is possible?

Reply:
On Windows nulls aren't allowed but it is technically possible for a key to begin with an `=` (e.g. `=key=value`). However this is essentially a backwards compatibility hack used only by `cmd.exe` so as to have hidden variables for internal use. It is not documented and is not meant to be set or used by other code. That said, nothing actually prevents a program from doing so.

Reply:
There are also some targets where the env is literally a hashmap. Sgx, for example. I'm not sure in what sense it's an environment at that point, but perhaps I'm missing something.